### PR TITLE
add pombot, using chatter

### LINKED ISSUE #115: implement basic development bot "pombot-jenn"
Should be able to join the #test channel and respond to the prompt `pom help`.

### LINKED ISSUE #117: support command `pombot help`
it should know about the following (stubbed out) commands:
- start
- stop
- i will
- status

### PR DESCRIPTION
**Pull request by jennschiffer**
closes #115, #117

creates a bot `pombot` with the command triggered by entering `pom help`.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ module.exports = function(grunt) {
           'src/robocoup/index.js',
           'src/robocoup/commands/expertise.js',
           'src/robocoup/commands/help.js',
+          'src/new-bot/**/*.js',
```
Comment (cowboy) [inline Gruntfile.js]: This should be the directory of pombot.

File: Gruntfile.js [hunk 2]
```diff
@@ -78,6 +79,11 @@ module.exports = function(grunt) {
         files: ['.eslintrc*', 'eslint/*'],
         tasks: ['eslint'],
       },
+      // This makes working on chatter with npm link SO much easier!
```
Comment (cowboy) [inline Gruntfile.js]: This "framework" Gruntfile section can be omitted.

File: eslint/eslint-es2015.yaml [hunk 3]
```diff
@@ -10,6 +10,7 @@ rules:
```
Comment (cowboy) [inline eslint/eslint-es2015.yaml]: Perhaps the eslint stuff should be a separate commit.

File: src/pombot/commands/help.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*
+ * A command to show help information.
+ */
+import {createCommand} from 'chatter';
+
+export default createCommand({
```
Comment (cowboy) [inline src/pombot/commands/help.js]: You don't need to implement help, `createCommand` does that automatically.

File: src/pombot/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    rtmClient: new RtmClient(config.tokens.pombot, {
+      dataStore: new MemoryDataStore(),
+      autoReconnect: true,
+    }),
+    webClient: new WebClient(config.tokens.pombot),
+  },
+  createMessageHandler() {
+    return createConversation([
+      this.createSlackMessageHandler({dm: true}, [
+        helpCommand
```
Comment (cowboy) [inline src/pombot/index.js]: There should be a top-level command with the name `pom` that has a description but otherwise just aggregates sub-commands.

Comment (cowboy) [inline src/pombot/index.js]: In public channels, give it the name `pom`. In DM, omit the name and it will respond to all messages.

File: src/pombot/commands/help.js (hunk 4 above)
Comment (jennschiffer) [inline src/pombot/commands/help.js]: oh so you mean "help" is a reserve command?

File: src/pombot/index.js (hunk 5 above)
Comment (cowboy) [inline src/pombot/index.js]: DM command:
https://github.com/bocoup/bot-framework/blob/483cdd6b54b5f8d9b70399c0535fde530d5c4da5/examples/slack.js#L126-L136

Public command:
https://github.com/bocoup/bot-framework/blob/483cdd6b54b5f8d9b70399c0535fde530d5c4da5/examples/slack.js#L141-L149

Notice they're the same, except the public one has a name.

File: src/pombot/commands/help.js (hunk 4 above)
Comment (cowboy) [inline src/pombot/commands/help.js]: https://github.com/bocoup/bot-framework/blob/483cdd6b54b5f8d9b70399c0535fde530d5c4da5/src/message-handler/command.js#L107-L120

Comment (cowboy) [inline src/pombot/commands/help.js]: If you define a `help` command, the built-in one will never run. But you want the built-in one to run, because it does all the hard work for you! As long as you give commands a `name` and `description` and optional `usage`, the built-in `help` will do its thing.

Comment (cowboy) [inline src/pombot/commands/help.js]: And `name` is optional for the top-level command. You only want to give the top-level command a name if you don't want `help` potentially displaying for every message. Like when the command is in a public channel.

File: package.json [hunk 6]
```diff
@@ -2,13 +2,16 @@
...
     "aws-sdk": "^2.3.5",
-    "babel-cli": "^6.4.5",
+    "babel-cli": "^6.8.0",
+    "babel-core": "^6.8.0",
     "babel-plugin-transform-runtime": "^6.4.0",
     "babel-preset-es2015": "^6.3.13",
     "babel-register": "^6.4.3",
     "babel-runtime": "^6.3.19",
     "bluebird": "^3.1.1",
+    "chatter": "^0.1.0",
```
Comment (cowboy) [inline package.json]: Please lock this to `"0.1.0"` instead of `"^0.1.0"`

File: src/pombot/commands/iwill.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/*
+ * A stub for the "i will" command.
+ */
+import {createCommand} from 'chatter';
+
+export default createCommand({
+  name: 'i will',
```
Comment (cowboy) [inline src/pombot/commands/iwill.js]: Does this actually work? I haven't tried commands with spaces in them!

**Comment by cowboy** [comment]
The only comment I have at this point is that the eslint changes should be in a separate commit. Other than that (and my other in-line comments) 👍

File: src/pombot/commands/iwill.js (hunk 7 above)
Comment (jennschiffer) [inline src/pombot/commands/iwill.js]: it does!

File: eslint/eslint-es2015.yaml (hunk 3 above)
Comment (jennschiffer) [inline eslint/eslint-es2015.yaml]: ok separated it out and squashed the rest!

File: src/pombot/commands/iwill.js (hunk 7 above)
Comment (cowboy) [inline src/pombot/commands/iwill.js]: OMG i'm so 1337

File: config.js [hunk 8]
```diff
@@ -7,11 +7,12 @@ module.exports = {
   tokens: {
     robocoup: process.env.TOKEN_ROBOCOUP,
     thanksbot: process.env.TOKEN_THANKSBOT,
+    pombot: process.env.TOKEN_POMBOT,
   },
   email: {
     key: process.env.AWS_ACCESS_KEY_ID,
     secret: process.env.AWS_SECRET_ACCESS_KEY,
-    region: process.env.AWS_DEFAULT_REGION || 'us-east-1'
+    region: process.env.AWS_DEFAULT_REGION || 'us-east-1',
```
Comment (cowboy) [inline config.js]: This should be in a separate commit. It's an unrelated-to-the-PR linting fix.

**Comment by cowboy** [comment]
Other than my one final comment, 👍
